docs/references/glossary.md

```diff
@@ -0,0 +1,310 @@
+
+## Permission Rule
+
+Rules are predicate-based controls that tap into a [resource](#permission-resource)'s data.
+
+## Persona
+
+Alternative term for a [User Role](#user-role).
+
+## Plugin
+
+A module in Backstage that adds a feature. All functionality in Backstage, even the core features, are implemented as plugins.
+
+## Policy
+
+User [permissions](#permission) are authorized by a central, user-defined [permission](#permission) policy. At a high level, a policy is a function that receives a Backstage user and [permission](#permission), and returns a decision to allow or deny. Policies are expressed as code, which decouples the framework from any particular [authorization](#authorization) model, like role-based access control (RBAC) or attribute-based access control (ABAC).
+
+## Policy decision
+
+Two important responsibilities of any authorization system are to decide if a user can do something, and to enforce that decision. In the Backstage permission framework, policies are responsible for decisions and plugins (typically backends) are responsible for enforcing them.
+
+## Popup
+
+A separate browser window opened on top of the previous one.
+
+## Procedure
+
+A set of actions that accomplish a goal, usually as part of a [use case](#Use-Case). A procedure can be high-level, containing other procedures, or can be as simple as a single [task](#Task).
+
+## Query Translators
+
+An abstraction layer between a search engine and the [Backstage Search](#search) backend. Allows for translation into queries against your search engine. Part of [Backstage Search](#search).
+
+## Refresh Token
+
+A string that an [OAuth](#oauth) client can use to get a new access token.
+
+https://oauth.net/2/refresh-tokens/
+
+## Resource
+
+In the Backstage Catalog, an [entity](#entity) that represents a piece of physical or virtual infrastructure, for example a database, required by a component.
+
+https://backstage.io/docs/features/software-catalog/system-model
+
+## Role
+
+See [User Role](#User-Role).
+
+## Scope
+
+A string that describes a certain type of access that can be granted to a user using OAuth.
+
+## Search
+
+A Backstage plugin that provides a framework for searching a Backstage [app](#app), including the [Software Catalog](#Software-Catalog) and [TechDocs](#TechDocs). A core feature of Backstage.
+
+## Search Engines
+
+Existing search technology that [Backstage Search](#search) can take advantage of through its modular design. Lunr is the default search in [Backstage Search](#search). Part of [Backstage Search](#search).
+
+## Software Catalog
+
+A Backstage plugin that provides a framework to keep track of ownership and metadata for any number and type of software [components](#component). A core feature of Backstage.
+
+## Software Templates
+
+A Backstage plugin with which to create [components](#component) in Backstage. A core feature of Backstage.
```

Comment:
The tagline could be tweaked a bit down the line here :)

This is also a good opportunity to add the Scaffolder term and cross linking the two, since we confusingly refer to both in different contexts.